src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -168,8 +168,8 @@ public void onNewInfo(ClusterInfo info) {
                     warnAboutDiskIfNeeded(entry);
                     if (entry.getFreeBytes() < DiskThresholdDecider.this.freeBytesThresholdHigh.bytes() ||
                             entry.getFreeDiskAsPercentage() < DiskThresholdDecider.this.freeDiskThresholdHigh) {
-                        if ((System.currentTimeMillis() - lastRun) > DiskThresholdDecider.this.rerouteInterval.millis()) {
-                            lastRun = System.currentTimeMillis();
+                        if ((TimeValue.nsecToMSec(System.nanoTime() - lastRunNS)) > DiskThresholdDecider.this.rerouteInterval.millis()) {
```

Comment:
This could also be

```java
if ((System.nanoTime() - lastRunNS) > DiskThresholdDecider.this.rerouteInterval.nanos()) {
```

To avoid an extra time value conversion